lib/tasks/hathi_etas.rake

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+HATHI_DATA_HOME = '/data/hathitrust_data'
+
+namespace :hathitrust do
+  desc 'Process overlap file for emergency access to restricted HathiTrust material'
+  task :process_hathi_etas, [:hathi_full, :overlap_psu] do |_task, args|
+    Dir.chdir(HATHI_DATA_HOME.to_s) do
+      Rake::Task['hathitrust:pare_hathi_full'].invoke(args[:hathi_full])
+      Rake::Task['hathitrust:extract_overlap_oclc'].invoke(args[:overlap_psu])
+      Rake::Task['hathitrust:filter_overlap'].invoke
+      Rake::Task['hathitrust:extract_uniq_oclc'].invoke
+    end
+  end
+
+  desc 'Pare the monthly file down to just the needed data: Hathi Trust id and OCLC number'
+  task :pare_hathi_full, [:hathi_full] do |_task, args|
+    print `gunzip -c #{args[:hathi_full]} | \
+               csvcut -t -c 1,8 -z 1310720 | \
+               csvgrep -c 1,2 -r ".+" | \
+               sort | uniq > hathi_full_dedupe.csv`
+  end
+
+  desc 'Extract the unique set of OCLC numbers from the overlap report'
+  task :extract_overlap_oclc, [:overlap_psu] do |_task, args|
+    print `csvgrep -t -c 4 -r ".+" #{args[:overlap_psu]} | \
```

Comment:
same here we could curl it. If we knew the filename on the HT server we could use that as part of a formula for the curl operation.

Reply:
yes, I think so, I will give it a try

Reply:
not sure where the overlap file is on the HT side, there is the shared box folder but I do not know if we can find it elsewhere 🤔

Reply:
well lets ask a follow up question to HT when they write back to us

Reply:
we talked about this on slack, we will handle moving the overlap file manually.